lib/diagnostic.js

```diff
@@ -1,13 +1,26 @@
 'use strict';
 
+
+//comfort:0
+//clarity: 1.7
+//Used google and class notes
+
+
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function () {                       //object min is a function (which is the minimum in the array)
+  let reduce = function (array, callback) {       //variable reduce is a function which reduces the array
+    for (i = 0, i < array.length; i++) {  //loops through the array
+    callback(reduce, e, array[i], a);    //uses callback to reduce to minimum #
+  });
+}
+  return array.min;
 };
```

Comment:
You have some of the right ideas here, but you are confused about your variables are, and how callbacks work. Please take a look at this solution below. Note that it uses a ternary expression, which is really just a 1-liner if statement. 
```
const min = function (previous, current) {
  return previous < current ? previous : current;
};
```